clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -284,8 +282,8 @@ public void seek(TopicPartition tp, long offset) {
         return this.assignment.partitionSet();
     }
 
-    public List<TopicPartition> fetchablePartitions() {
-        List<TopicPartition> fetchable = new ArrayList<>();
+    public Set<TopicPartition> fetchablePartitions() {
+        Set<TopicPartition> fetchable = new HashSet<>();
```

Comment:
Yeah, my fault for suggesting this earlier. I had forgotten about the significance of the order. Instead of a predicate, we could also pass in a set of the partitions to exclude. Switching to `removeAll` also sounds better.